app/helpers/automated_tests_helper.rb

```diff
@@ -38,18 +38,21 @@ def update_test_groups_from_specs(assignment, test_specs)
     test_specs_path = assignment.autotest_settings_file
     # create/modify test groups based on the autotest specs
     test_group_ids = []
-    test_specs['testers'].each do |tester_specs|
+    criteria_map = assignment.ta_criteria.pluck(:type, :name, :id).map do |type, name, id_|
+      ["#{type}:#{name}", id_]
+    end.to_h
+    test_specs['testers']&.each do |tester_specs|
       next if tester_specs['test_data'].nil?
 
-      tester_specs['test_data'].each do |test_group_specs|
+      tester_specs['test_data']&.each do |test_group_specs|
         test_group_specs['extra_info'] ||= {}
         extra_data_specs = test_group_specs['extra_info']
         test_group_id = extra_data_specs['test_group_id']
         display_output = extra_data_specs['display_output'] || TestGroup.display_outputs.keys.first
         test_group_name = extra_data_specs['name'] || TestGroup.model_name.human
         criterion_id = nil
         unless extra_data_specs['criterion'].nil?
-          criterion_id = extra_data_specs['criterion']
+          criterion_id = criteria_map[extra_data_specs['criterion']]
```

Comment:
If we can't find the criterion id here then show a warning message that the criterion can't be found